src/node_worker.cc

```diff
@@ -918,6 +926,63 @@ void GetEnvMessagePort(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+void SetMainPort(const FunctionCallbackInfo<Value>& args) {
+  if (args.Length() == 0 || args[0]->IsNullOrUndefined()) {
+    main_port = nullptr;
+    return;
+  }
+
+  MessagePort* port;
+  ASSIGN_OR_RETURN_UNWRAP(&port, args[0]);
+
+  main_port = port;
+}
+
+void SendToWorker(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Local<Context> context = env->context();
+  v8::Isolate* isolate = args.GetIsolate();
+
+  // args[0] is the target worker thread id
+  // args[1] is the message to send via PostMessage
+  // args[2] is the transfer list to send via PostMessage, it is optional
+  CHECK_GT(args.Length(), 1);
+
+  CHECK(args[0]->IsUint32());
+  auto tid = args[0].As<v8::Uint32>()->Value();
+
+  MessagePort* port;
+
+  if (tid == 0) {
+    port = main_port;
+  } else {
+    auto it = workers.find(tid);
+    if (it == workers.end()) {
+      std::string message = SPrintF("Invalid worker id %d", tid);
+      THROW_ERR_WORKER_INVALID_ID(isolate, message.c_str());
+      return;
+    }
+
+    port = it->second->parent_port();
+  }
+
+  TransferList transfer_list;
+  // Not all messages require a transfer_list
+  if (args.Length() > 2) {
+    if (!GetTransferList(env, context, args[2], &transfer_list)) {
+      return;
```

Comment:
Should this throw like in all other cases where `port->PostMessage()` is not reached?
Otherwise caller will wait for an answer which for sure will never come.

Reply:
It's not needed: `GetTransferList` already throws. It returns false only if the list was empty.

Reply:
If I understand this correct an empty transferlist would result in a hanging caller as no exception is thrown and no message is posted.

Reply:
No, why? The empty transferlist is supported AFAIK.